src/Bindings/ManualBindings_World.cpp

```diff
@@ -621,6 +665,38 @@ static int tolua_cWorld_ScheduleTask(lua_State * tolua_S)
 
 
 
+static int tolua_cWorld_SpawnSplitExperienceOrbs(lua_State* tolua_S)
+{
+	cLuaState L(tolua_S);
+	if (
+		!L.CheckParamSelf("cWorld") ||
+		!L.CheckParamNumber(2, 5) ||
+		!L.CheckParamEnd(6)
+	)
+	{
+		return 0;
+	}
+
+	cWorld * self = nullptr;
+	double X = 0;
+	double Y = 0;
+	double Z = 0;
+	int Reward = 0;
+	L.GetStackValues(1, self, X, Y, Z, Reward);
```

Comment:
Is it ok to port API to lua with different argument list ? will this lead to some confusion?

Reply:
I am really confused here, I don't know how should I write method argument description for DoWithNearestPlayer, according to `tolua_cWorld_DoWithNearestPlayer`,  it get parameter x,y,z from input 2 - 5, should I descript parameter as number rather than vector3d?

Reply:
1. It's okay to have different signatures in the Lua API but if you want to change the functions to use `Vector3d` on the C++ side then that would be even better.

2. The Lua docs should always match the bindings so if the bindings take a vector then the docs should say it takes a vector.

Reply:
I'm realising now that your second comment is probably referring to the `CheckParam` part, in which case the way to do it would be with `CheckParamUserType`:

```cpp

// Check params:
cLuaState L(tolua_S);
if (
	!L.CheckParamSelf("cWorld") ||
	!L.CheckParamUserType(2, "Vector3<double>") ||
    // Other params ...
)
{
	return 0;
}
```